discovery-aws-api/src/main/scala/akka/discovery/awsapi/ecs/EcsSimpleServiceDiscovery.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.discovery.awsapi.ecs
+
+import java.net.{ InetAddress, NetworkInterface }
+import java.util.concurrent.TimeoutException
+
+import akka.actor.ActorSystem
+import akka.discovery.SimpleServiceDiscovery
+import akka.discovery.SimpleServiceDiscovery.{ Resolved, ResolvedTarget }
+import akka.discovery.awsapi.ecs.EcsSimpleServiceDiscovery._
+import akka.pattern.after
+import com.amazonaws.services.ecs.model.{ DescribeTasksRequest, DesiredStatus, ListTasksRequest, Task }
+import com.amazonaws.services.ecs.{ AmazonECS, AmazonECSClientBuilder }
+
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+import scala.collection.immutable.Seq
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+class EcsSimpleServiceDiscovery(system: ActorSystem) extends SimpleServiceDiscovery {
+
+  override def lookup(name: String, resolveTimeout: FiniteDuration): Future[Resolved] = {
+    val ecsClient = AmazonECSClientBuilder.defaultClient()
+    import system.dispatcher
+    Future.firstCompletedOf(
+      Seq(
+        after(resolveTimeout, using = system.scheduler)(
+          Future.failed(new TimeoutException("Future timed out!"))
+        ),
+        Future {
+          Resolved(
+            serviceName = name,
+            addresses = for {
+              task <- resolveTasks(ecsClient, name)
+              container <- task.getContainers.asScala
+              networkInterface <- container.getNetworkInterfaces.asScala
+            } yield
+              ResolvedTarget(
+                host = networkInterface.getPrivateIpv4Address,
+                port = None
+              )
+          )
+        }
+      )
+    )
+  }
+}
+
+object EcsSimpleServiceDiscovery {
+
+  // InetAddress.getLocalHost.getHostAddress throws an exception when running
+  // in awsvpc mode because the container name cannot be resolved.
+  // ECS provides a metadata file
+  // (https://docs.aws.amazon.com/AmazonECS/latest/developerguide/container-metadata.html)
+  // that we ought to be able to use instead to find our IP address, but the
+  // metadata file does not get set when running on Fargate. So this is our
+  // only option for determining what the canonical Akka and akka-management
+  // hostname values should be set to.
+  def getContainerAddress: Either[String, InetAddress] =
```

Comment:
should get some documentation for users why they'd use this -- (i.e. explain how you used it in the demo app for example)

Reply:
I included something in the main discovery docs: https://github.com/akka/akka-management/pull/140/files#diff-2f9309809fa0ed116cc955a91dc2c3b0R439

Or do you mean the source comment should be clearer?